test_chess.py

```diff
@@ -34,3 +34,113 @@ def test_square(self):
         for rank in board._board:
             for piece in rank:
                 assert board[board.square_of(piece)] is piece
+
+
+class TestSquare:
+    """Unit tests exclusive to the squares."""
+
+    def test_square_operations(self):
+        """Test that squares can safely be operated on and that operations can be stacked."""
+        from src.square import Vector, Square
+
+        assert Square("e4") + Vector(-1, 0) == Square("e5")
+        assert Square("e4") + Vector(0, -1) == Square("d4")
+        assert Square("e4") + Vector(+1, 0) == Square("e3")
+        assert Square("e4") + Vector(0, +1) == Square("f4")
+
+
+class TestPiece:
+    """Unit tests for pieces."""
+
+    def test_legal_moves(self):
+        """Test whether the pieces generate proper legal moves. One example for each piece."""
+        from src.square import Square
+        from src.pieces import Color, Pawn, King, Knight, Rook, Bishop, Queen
+
+        def condition(square: Square) -> bool:
+            """Draft condition closure for `legal_moves`."""
+            return square.rank >= 0 and square.rank < 8 and square.file >= 0 and square.file < 8
```

Comment:
```py
return all((0 <= square.rank, square.file < 8))
```
or
```py
return 0 <= square.rank < 8 and 0 <= square.file < 8
```

Reply:
The second one is very elegant, I'll take it.